controllers/machinedeletionremediation_controller.go

```diff
@@ -576,6 +588,36 @@ func (r *MachineDeletionRemediationReconciler) getMachineOwnerNodes(ctx context.
 	return machineOwnerNodes, nil
 }
 
+func (r *MachineDeletionRemediationReconciler) getMachineNameNsFromCR(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation) (machineName, machineNs string, err error) {
+	var node *v1.Node
+	node, err = r.getNodeFromCR(ctx, remediation)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			r.Log.Error(err, nodeNotFoundErrorMsg, "node name", remediation.Name)
+			err = nodeNotFoundError
+		}
+		return "", "", err
+	}
+
+	machineName, machineNs, err = getMachineNameNsFromNode(node)
+	if err != nil {
+		r.Log.Error(err, "could not get Machine Name NS from Node", "node", node.Name, "annotations", node.GetAnnotations())
+		return "", "", unrecoverableError
+	}
+	return machineName, machineNs, nil
+}
+
+func (r *MachineDeletionRemediationReconciler) getMachineNameNsFromOwnerReference(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation) (machineName, machineNs string) {
+	for _, owner := range remediation.GetOwnerReferences() {
+		if owner.Kind == machineKind {
```

Comment:
Nit: I don't see a value of using a variable here over the hardcoded string "Machine".
IMO the string here is more readable since I don't need to lookup the var value and there is no value for using the var since it's only used once here.